src/Common/MemoryBufferWriter.cs

```diff
@@ -170,5 +216,66 @@ public byte[] ToArray()
 
             return result;
         }
+
+        public override void Flush() { }
+        public override Task FlushAsync(CancellationToken cancellationToken) => Task.CompletedTask;
+        public override int Read(byte[] buffer, int offset, int count) => throw new NotSupportedException();
+        public override long Seek(long offset, SeekOrigin origin) => throw new NotSupportedException();
+        public override void SetLength(long value) => throw new NotSupportedException();
+
+        public override void WriteByte(byte value)
+        {
+            if (_currentSegment != null && (uint)_position < (uint)_currentSegment.Length)
+            {
+                _currentSegment[_position] = value;
+            }
+            else
+            {
+                AddSegment();
+                _currentSegment[0] = value;
+            }
+
+            _position++;
+            _bytesWritten++;
+        }
+
+        public override void Write(byte[] buffer, int offset, int count)
+        {
+            var position = _position;
+            if (_currentSegment != null && position < _currentSegment.Length - count)
+            {
+                Buffer.BlockCopy(buffer, offset, _currentSegment, position, count);
+
+                _position = position + count;
+                _bytesWritten += count;
+            }
+            else
+            {
+                BuffersExtensions.Write(this, buffer.AsSpan(offset, count));
+            }
+        }
+
+#if NETCOREAPP2_1
+        public override void Write(ReadOnlySpan<byte> span)
+        {
+            if (_currentSegment != null && span.TryCopyTo(_currentSegment.AsSpan().Slice(_position)))
```

Comment:
AsSpan takes arguments to do the slice at the same time.

Reply:
Ya I know.

Reply:
But then I have to pass the length

Reply:
_position is already length?

`_currentSegment.AsSpan(0, _position)`

Reply:
Ah, _position is where you start.

`_currentSegment.AsSpan(_position, _currentSegment.Length - _position)`?

Reply:
I didn't want to do that but I can.